### PR TITLE
Add a Procfile for easily running the app

### PR DESCRIPTION
**Pull request by cycomachead**
A Profile is a standard way of specifying resources an app needs to run.

With a Procfile, you can use a "runner" which automatically starts all the relevant commands, and will display output in a window.
These runners automatically source the `.env` file before running.

For example, Foreman is a ruby gem that runs procfiles and you can just do `foreman s` from within the snapCloud directory.
However, there's runners in every language basically.

Options:
Foreman: (ruby) https://github.com/ddollar/foreman
node-foreman: https://github.com/strongloop/node-foreman
shoreman: (bash) https://github.com/chrismytton/shoreman

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+app: lapis server
+db: postgres -D /usr/local/var/postgres
```
Comment (cycomachead) [inline Procfile]: This is the default mac location. I don't know what it is on other systems. I'm happy to make the $POSTGRES_DATA_DIR or something else if there's a different directly so this is usable by anyone.

**Comment by bromagosa** [comment]
Hmm, I don't understand what this solves. I'm guessing this is just another way to run the app and we'll still be using the regular daemon script strategy every Linux utility uses?

In that case, isn't it a bit confusing and redundant to have two different recommended ways to start it?

**Comment by cycomachead** [comment]
Sorry, I should have been more clear in the description. :)

In some cases, like if we were using Heroku or another similar service, the Procfile would be the way we run the app when deployed. In all other cases, it's used simply for development. 

The main problem the profile solves in running multiple resources required to have a functioning app. Initially, it was just lapis and postgres, but while working on the frontend, it's useful to have the Snippets watcher running. And just now, I added a development SMTP server to the Procfile.
A "fake" (or dev) SMTP service also allows anyone to get started more easily, and doesn't require an internet connection to see emails come on. 
Basically you get an output that shows you the status of all 4 processes in 1 terminal window. When you `^C` the runner, then all your processes are killed too. :)

Otherwise, while I've never used it, foreman specifically does include a utility to generate system daemons and other installers, in case you wanted to deploy the app in another setup, or say, include restarting the postgres server as part of restarting the app.

File: Procfile (hunk 1 above)
Comment (bromagosa) [reply Procfile]: Not in Linux. This is what `SHOW data_directory;` spits out in the server:

`/var/lib/postgresql/9.5/main`

**Comment by bromagosa** [comment]
This looks great, but the postgres data directory is not the same across OSes. If it needs to work in a single system, it should be the one where it's deployed, otherwise let's add another envvar to specify it.

**Comment by cycomachead** [comment]
OK, makes sense. I still need an environment variable to be an option, otherwise the file won't be useful locally, where it will be used. But I created one that defaults to the directory on the production machine.
